### DISCUSSION TITLE
Identify the server module in the client bundle

### DISCUSSION THREAD

**Discussion by ram2104** [remix-run/remix#3428] [Q&A]
I was migrating one of the React, Redux, and React-Router-based app to Remix.  Since my app was based out of the react-router I thought I can incrementally migrate my application to remix.

For the migration, I'm referring to this [document](https://remix.run/docs/en/v1/guides/migrating-react-router-app). I have almost come to the end of the suggested migration steps. 

When I'm trying to verify migrated app in the browser. It's failing to load the screen because it says :

`Uncaught Error: Cannot initialize 'routeModules'. This normally occurs when you have server code in your client modules.`
Check this link for more details: **https://remix.run/pages/gotchas#server-code-in-client-bundles**

Is there any way to identify the server code in the client bundle? I can't do it manually because it has a dependency on the 40-50 module.

Need suggestions on how to proceed further.
struggling from long time

**Comment by lili21** [comment]
I think we need a way to analyze the bundle result. Esbuild provide the `--metafile` option to output  information about the bundle, and the we can use https://bundle-buddy.com/ to do the rest wok.

**Reply by kiliman** [reply]
I created a patch you can add to your project that will generate the metafile automatically.

https://github.com/kiliman/remix-esbuild-analysis
